### ISSUE TITLE
`tvae` inference indexes the latent placeholder by batch size instead of sequence position

### ISSUE THREAD

**Issue by p3zo** [p3zo/rhythmic-relationships#17]
```python
# scripts/modeling/tvae/train.py:52-63
latent_placeholder = torch.zeros(n_new_tokens + len(y), 1, latents.size(-1), device=DEVICE)

for _ in range(n_new_tokens):
    latent_placeholder[len(y) - 1, 0, :] = latents[len(y)]
    dec_seg_emb = latent_placeholder[: len(y), :]
```

`y` has shape `(B, T)`, so `len(y)` is the **batch size**, not the sequence position. And `y` grows along dim 1 (`torch.cat(..., dim=1)` at `:81`), so `len(y)` never changes across the loop.

Consequences:
- the same row `B-1` is overwritten every iteration; every other row of the placeholder stays zero
- `latents[len(y)]` indexes the latent tensor by batch size, which is unrelated to the token being generated
- the decoder context is a fixed `(B, 1, d)` slice for the whole generation

The MuseMorphose code this is adapted from indexes by the number of generated tokens.

**Fix**: track the generated position explicitly (`for t in range(n_new_tokens): latent_placeholder[t] = latents[t]`) and slice the placeholder by `t + 1`, with the batch dimension kept separate from the time dimension.
